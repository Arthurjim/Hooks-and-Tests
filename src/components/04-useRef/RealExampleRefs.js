import React,{useState} from 'react'
import '../03-examples/exampe..css'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'
const RealExampleRefs = () => {
    const [show, setShow] = useState(false)
    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr/>
            {show && <MultipleCustomHooks />}
            <button className="btn btn-primary mt-5"
            onClick={()=>{setShow(!show)}}>
                {show ? 'Hide' : 'Show'}
            </button>
        </div>
    )
}

export default RealExampleRefs