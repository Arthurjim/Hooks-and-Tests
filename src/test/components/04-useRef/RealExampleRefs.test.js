import { shallow } from "enzyme"
import RealExampleRefs from '../../../components/04-useRef/RealExampleRefs'
describe('Preubas en el componente RealExampleRefs', () => {
    const wrapper = shallow(<RealExampleRefs />)
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

        
    })

    test('Debe de mostrar el componenet <MultipleCustomHook />', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
        
    })
    
    
})
