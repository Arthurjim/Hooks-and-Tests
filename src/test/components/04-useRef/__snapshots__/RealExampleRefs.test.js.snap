// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas en el componente RealExampleRefs Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
